File: app/src/main/java/vergecurrency/vergewallet/view/ui/activity/SplashActivity.java
package vergecurrency.vergewallet.view.ui.activity;

import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;

import java.io.IOException;
import java.security.GeneralSecurityException;

import vergecurrency.vergewallet.helpers.utils.LanguagesUtils;
import vergecurrency.vergewallet.helpers.utils.UIUtils;
import vergecurrency.vergewallet.view.base.BaseActivity;
import vergecurrency.vergewallet.R;
import vergecurrency.vergewallet.service.model.PreferencesManager;
import vergecurrency.vergewallet.view.ui.activity.firstlaunch.FirstLaunchActivity;
import vergecurrency.vergewallet.wallet.WalletManager;

public class SplashActivity extends BaseActivity {
	//DbOpenHelper dbOpenHelper;
	//private AbstractDaoSession daoSession;


	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		//TestFairy token for getting info on beta testing
		//TestFairy.begin(this, "a67a4df6e2a8a0c981638eb0f168297fd45aed73");
		//init db
		//dbOpenHelper = new DbOpenHelper(this,"verge.db",1);
		//Database db = dbOpenHelper.getWritableDb();
		//daoSession = new AbstractDaoMaster;

		//gets the holy preferences


		try {
			PreferencesManager.initEncryptedPreferences(this);
		} catch (GeneralSecurityException e) {
			e.printStackTrace();
		} catch (IOException e) {
			e.printStackTrace();
		}

		setContentView(R.layout.activity_splash);

		//Just to have the splash screen going briefly
		new Handler().postDelayed(this::startApplication, 2000);
	}

	private Context updateBaseContextLocale(Context context) {
		return LanguagesUtils.setLocale(context, PreferencesManager.getCurrentLanguage());
	}


	public void startApplication() {

		if (PreferencesManager.getFirstLaunch()) {
			startActivity(new Intent(getApplicationContext(), FirstLaunchActivity.class));
			finish();
		} else {
			try {
				WalletManager.startWallet();
			} catch (Exception e) {
				Toast.makeText(getApplicationContext(), "Something went wrong", Toast.LENGTH_SHORT).show();
			}
			startActivity(new Intent(getApplicationContext(), WalletActivity.class));
			finish();
		}
	}


	@Override
	public void onBackPressed() {
	}

}

File: app/src/main/kotlin/vergecurrency/vergewallet/view/ui/activity/SplashActivity.kt
package vergecurrency.vergewallet.view.ui.activity

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.widget.Toast
import vergecurrency.vergewallet.R
import vergecurrency.vergewallet.service.model.EncryptedPreferencesManager
import vergecurrency.vergewallet.service.model.PreferencesManager
import vergecurrency.vergewallet.service.model.network.layers.TorManager
import vergecurrency.vergewallet.view.base.BaseActivity
import vergecurrency.vergewallet.view.ui.activity.firstlaunch.FirstLaunchActivity
import vergecurrency.vergewallet.wallet.WalletManager

class SplashActivity : BaseActivity() {
    //DbOpenHelper dbOpenHelper;
    //private AbstractDaoSession daoSession;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        //init db
        //dbOpenHelper = new DbOpenHelper(this,"verge.db",1);
        //Database db = dbOpenHelper.getWritableDb();
        //daoSession = new AbstractDaoMaster;
        setContentView(R.layout.activity_splash)
        //Just to have the splash screen going briefly
        Handler().postDelayed({ r.run() }, 500)
    }

    private var r: Runnable = Runnable {
        try {
            TorManager.getInstance(applicationContext)

            //TODO TESTING only
            EncryptedPreferencesManager.usingTor = true

            if (EncryptedPreferencesManager.usingTor) {


                TorManager.getInstance(null).execute()

            }
        } catch (e: java.lang.Exception) {
            e.printStackTrace()
            EncryptedPreferencesManager.usingTor = false
            Toast.makeText(applicationContext, "Tor could not launch. Switching to Clearnet. ALPHA ONLY.", Toast.LENGTH_LONG).show()
        }


        if (PreferencesManager.isFirstLaunch) {
            startActivity(Intent(applicationContext, FirstLaunchActivity::class.java))
            finish()
        } else {
            try {
                //init took place in VergeWalletApplication
                WalletManager.startWallet()
            } catch (e: Exception) {
                Toast.makeText(applicationContext, "Something went wrong", Toast.LENGTH_SHORT).show()
            }

            startActivity(Intent(applicationContext, WalletActivity::class.java))
            finish()
        }

    }

    override fun onBackPressed() {}
}
